fhir-persistence/src/test/java/com/ibm/fhir/persistence/test/common/AbstractIncludeRevincludeTest.java

```diff
@@ -385,6 +475,44 @@ else if (resource instanceof Patient) {
         }
     }
 
+    /**
+     * This test queries a Device with _elements search parameter and requests the reverse inclusion of
+     * an Observation that references the Device.
+     * The Device has an Observation that references it, and only the Device resource has the
+     * _elements processing applied to it.
+     * @throws Exception
+     */
+    @Test
+    public void testRevIncludeWithElements() throws Exception {
+        Map<String, List<String>> queryParms = new HashMap<String, List<String>>();
+        queryParms.put("_id", Collections.singletonList(savedDevice1.getId()));
+        queryParms.put("_elements", Collections.singletonList("patient"));
+        queryParms.put("_revinclude", Collections.singletonList("Observation:focus"));
+        List<Resource> resources = runQueryTest(Device.class, queryParms);
+        assertNotNull(resources);
+        assertEquals(2, resources.size());
+        for (Resource resource : resources) {
+            if (resource instanceof Observation) {
+                assertEquals(savedObservation6.getId(), resource.getId());
+                // check elements are not filtered out
+                assertEquals("Device/" + savedDevice1.getId() + "/_history/1", resource.as(Observation.class).getSubject().getReference().getValue());
+                assertEquals("Device/" + savedDevice1.getId() + "/_history/3", resource.as(Observation.class).getDevice().getReference().getValue());
+                assertEquals("Device/" + savedDevice1.getId() + "/_history/2", resource.as(Observation.class).getFocus().get(0).getReference().getValue());
+                assertEquals("Specimen/1", resource.as(Observation.class).getSpecimen().getReference().getValue());
+            }
+            else if (resource instanceof Device) {
```

Comment:
note the pattern of continued lines, not line breaks 

} else if ... 

we can get that next time we're in the code.

Reply:
Done.